client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -0,0 +1,517 @@
+					countryCode: 'US',
+				},
+			},
+		];
+		const paymentDetails = {
+			total: {
+				label: translate( 'Total' ),
+				amount: {
+					currency: cart.currency,
+					value: cart.total_cost + '',
+				},
+			},
+			displayItems: cart.products.map( product => {
+				return {
+					label: product.product_name,
+					amount: {
+						currency: product.currency,
+						value: product.cost + '',
+					},
+				};
+			} ),
+		};
+
+		const paymentRequest = new PaymentRequest(
+			supportedPaymentMethods,
+			paymentDetails,
+			PAYMENT_REQUEST_OPTIONS
+		);
+
+		const environment = 'production' === config( 'env' ) ? undefined : 'sandbox';
+
+		paymentRequest.onmerchantvalidation = event => {
+			wpcom
+				.undocumented()
+				.applePayMerchantValidation( event.validationURL, environment )
+				.then( json => event.complete( json ) )
+				.catch( error => {
+					debug( 'onmerchantvalidation error' );
+					debug( error );
+				} );
+		};
+
+		return paymentRequest;
+	};
+
+	/**
+	 * @return {PaymentRequest} A configured payment request object.
+	 */
+	getPaymentRequestForBasicCard = () => {
+		const { cart, translate } = this.props;
+
+		const supportedPaymentMethods = [
+			{
+				supportedMethods: WEB_PAYMENT_BASIC_CARD_METHOD,
+				data: {
+					supportedNetworks: SUPPORTED_NETWORKS,
+				},
+			},
+		];
+		const paymentDetails = {
+			total: {
+				label: translate( 'Total' ),
+				amount: {
+					currency: cart.currency,
+					value: cart.total_cost,
+				},
+			},
+			displayItems: cart.products.map( product => {
+				return {
+					label: product.product_name,
+					amount: {
+						currency: product.currency,
+						value: product.cost,
+					},
+				};
+			} ),
+		};
+
+		return new PaymentRequest( supportedPaymentMethods, paymentDetails, PAYMENT_REQUEST_OPTIONS );
+	};
+
+	/**
+	 * @param {string} paymentMethod  Payment method.
+	 * @param {DOMEvent} event        Button even.
+	 */
+	submit = ( paymentMethod, event ) => {
+		event.persist();
+		event.preventDefault();
+
+		switch ( paymentMethod ) {
+			case WEB_PAYMENT_APPLE_PAY_METHOD:
+				{
+					const paymentRequest = this.getPaymentRequestForApplePay();
+
+					try {
+						paymentRequest
+							.show()
+							.then( paymentResponse => {
+								const { payerName, details } = paymentResponse;
+								const { token } = details;
+
+								if ( 'EC_v1' !== token.paymentData.version ) {
+									return; // Not supported yet.
+								}
+
+								this.props.transaction.payment = newCardPayment( {
+									tokenized_payment_data: token.paymentData,
+									name: payerName,
+									country: this.state.country,
+									'postal-code': this.state.postalCode,
+									card_brand: token.paymentMethod.network,
+								} );
+
+								setPayment( this.props.transaction.payment );
+
+								this.props.onSubmit( event );
+
+								paymentResponse.complete( 'success' );
+							} )
+							.catch( error => {
+								debug( 'show error' );
```

Comment:
These two debug statements could be combined, I think.

Reply:
Done, thanks!